Mb/development (#18)
hostmobility/documentation: docs/system/development.md

---
title: Software Development
tags:
  - HMX
  - MX-4
  - CT
  - MX-V
  - T30
  - T30 FR
  - C61
  - Development
---

## Overview

> Host Mobility Hardware run a Linux based system. As such, software is usually built from source. The target architechure is ARM32 and ARM64 which is different from typical desktops and servers so a cross-compiler is needed.

Reviewer: ```suggestion
Host Mobility computers run GNU/Linux where almost all software is built from source. The target architectures are ARM32 and ARM64. Since the typical development machines are x86_64, a cross-compiler is usually required.
```

[Software Development Kits (SDKs)](sdk.md) contain the toolchains needed to compile and debug applications for our systems.

## Yocto build system

> The operating system running on Host Mobility Hardware and is built using the Yocto.This combines software written by Host Mobility with free software from software and hardware developers all over the world. See [Yocto Build System](yocto.md) for details. 

Reviewer: ```suggestion
The reference operating system is built using Yocto. It combines software written by Host Mobility with free software from developers all over the world. See [Yocto Build System](yocto.md) for details. 
```
